source/non_matching/coupling.cc

```diff
@@ -238,7 +238,7 @@ namespace NonMatching
                 ocell->get_dof_indices(odofs);
 
                 // Reset the matrices.
-                cell_matrix = 0;
+                cell_matrix = 0.0;
```

Comment:
What is the error message you get? We have thousands of lines just like this, and the only thing I can see that's different here is that we have a `FullMatrix<typename Matrix::value_type>` and not just a `FullMatrix<double>`. Is the error related to the underlying scalar type?